cli/asc.js

```diff
@@ -704,27 +704,16 @@ exports.main = function main(argv, options, callback) {
         if (args.binaryFile.length) {
           let sourceMap = JSON.parse(wasm.sourceMap);
           sourceMap.sourceRoot = exports.sourceMapRoot;
-          sourceMap.sources.forEach((name, index) => {
-            let text = null;
-            if (name.startsWith(exports.libraryPrefix)) {
-              let stdName = name.substring(exports.libraryPrefix.length).replace(/\.ts$/, "");
-              if (exports.libraryFiles.hasOwnProperty(stdName)) {
-                text = exports.libraryFiles[stdName];
-              } else {
-                for (let i = 0, k = customLibDirs.length; i < k; ++i) {
-                  text = readFile(name.substring(exports.libraryPrefix.length), customLibDirs[i]);
-                  if (text !== null) break;
-                }
-              }
-            } else {
-              text = readFile(name, baseDir);
-            }
-            if (text === null) {
-              return callback(Error("Source file '" + name + "' not found."));
+          console.log(sourceMap.sources)
+          for (const index in sourceMap.sources) {
+            let internalPath = sourceMap.sources[index].replace(/\.ts$/, "");
+            let source = program.sources.find(source => source.internalPath === internalPath);
```

Comment:
Calls into the compiler without using a C-like API, hence is non-portable once we can compile the compiler itself to Wasm.

Reply:
Ah I had a feeling this was too easy.

Reply:
But wait. This already happened before: https://github.com/AssemblyScript/assemblyscript/blob/f9230cd2ab8a0044a844b2d626b742a3c3e86558/cli/asc.js#L522